### PR TITLE
Fixed Auto-Scraper and Added 3 different input modes

### LINKED ISSUE #7: Fix web scraping script to get Schema automatically

### PR DESCRIPTION
**Pull request by aivantg**
## What's in this PR

- Fixed Schema Scraper by switching from nightmare to puppeteer
- Added modes in settings: 'auto', 'auto-headless', and 'manual'

### Related Issues

closes #7

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -5,37 +5,36 @@ const { existsSync, readFileSync } = require('fs');
...
   const settings = JSON.parse(packageFile)['airtable-schema-generator'];
-  console.log('Found settings');
+  console.log('Found settings:');
   console.log(settings);
 
-  if (!settings || !settings.output || !settings.input || !settings.baseId) {
+  if (!settings || !settings.output || !settings.baseId || !settings.mode) {
     console.log(
-      "Couldn't find Input Folder Path, Output Folder Path and Base ID in Settings Object:"
+      "Couldn't find Input Folder Path, Output Folder Path, Mode, and Base ID in Settings Object:"
```
Comment (annieyro) [inline bin/index.js:21]: Nitpick: error message shouldn't always include "Input Folder"

File: bin/index.js [hunk 2]
```diff
@@ -69,31 +113,9 @@ function readSchemaFromFile(settings) {
...
-    );
-    console.log(
-      `Navigate to this page in the browser: ${baseUrl} \n\nRun this script in the console:\n\n${script}\n\nThis will save the output to your clipboard. Open up schemaRaw.json, delete everything, and paste!\n\n`
-    );
-  }
-
-  console.log('Retrieved Airtable Schema');
+  const schema = await getSchemaFromAirtable(settings);
 
+  console.log('Retrieived Airtable Schema');
```
Comment (annieyro) [inline bin/index.js]: nitpick: still spelled wrong 😂

File: bin/index.js [hunk 3]
```diff
@@ -59,6 +58,51 @@ function simplifySchema(schema) {
...
+        mode === 'auto'
+          ? 'Launching browser...'
+          : 'Launching headless browser...'
+      }`
+    );
+
+    // Load config vars
+    require('dotenv-safe').config({
+      path: '.airtable.env',
+      example: path.resolve(__dirname, '../.env.example')
```
Comment (annieyro) [inline bin/index.js:94]: Same comment from #15 - if we're going to call this `.airtable.env`, any thoughts on having the example file be called `.example.env` (or vice versa) ?

File: lib/scraper.js [hunk 4]
```diff
@@ -17,28 +16,30 @@ function getUrl(baseId) {
...
+    password
+  );
+  console.log('Submitting...');
+  await page.click(
+    '#sign-in-form-fields-root > div > label > input[type="submit"]'
+  );
+
+  await page.waitForSelector('.docs > .languageTabs > .tab');
+  console.log('Scraping schema from API page...');
+  const result = await page.evaluate(script);
```
Comment (annieyro) [inline lib/scraper.js:41]: 🙌 🙌 🙌  I look forward to the end of manual mode ❤️

**Review by annieyro — APPROVED** [review]
Awesome update! Small nitpicks (literally error messages, misspelling, and naming 😂 )

File: bin/index.js (hunk 1 above)
Comment (aivantg) [reply bin/index.js:21]: good catch, fixed in a future PR because i caught it there

File: bin/index.js (hunk 3 above)
Comment (aivantg) [reply bin/index.js:94]: fixed in future PR

File: bin/index.js (hunk 2 above)
Comment (aivantg) [reply bin/index.js]: HAHA must've gotten reverted somewhere

**Pull request merged**